src/Kunstmaan/MediaBundle/Helper/File/FileHandler.php

```diff
@@ -183,13 +200,28 @@ public function saveMedia(Media $media)
      */
     public function getOriginalFile(Media $media)
     {
-        $relativePath = sprintf(
-            '/%s.%s',
+        return $this->fileSystem->get($this->getFilePath($media), true);
+    }
+
+    /**
+     *
+     *
+     * @param Media $media
+     * @return string
+     */
+    private function getFilePath(Media $media)
+    {
+        $filename = $media->getOriginalFilename();
+
+        if (!empty($this->blacklistedExtensions)) {
+            $filename = preg_replace('/\.('.join('|', $this->blacklistedExtensions).')$/', '.txt', $filename);
+        }
+
+        return sprintf(
+            '/%s/%s',
```

Comment:
Please don't create a folder per file, but group them (ie. using a variant of the old OpenMercury naming scheme).

UUID : 12345678
file : original_filename.ext

=> /1/2/3/4/12345678-original_filename.ext (UUID has to be used here to prevent overwriting files)